src/layers/domain/core/cpm_system_id/v1.py

```diff
@@ -1,86 +1,124 @@
-from abc import abstractmethod
+import random
+import re
+from abc import ABC, abstractmethod
+from datetime import datetime
+from enum import StrEnum, auto
+from typing import Optional
 
 from domain.core.base import BaseModel
 
 
-class CpmSystemId(BaseModel):
+class CpmSystemIdType(StrEnum):
+    ASIDNUMBER = auto()
+    PARTYKEYNUMBER = auto()
+    PRODUCTID = auto()
+
+
+class CpmSystemId(BaseModel, ABC):
     key_name: str
-    latest: int
-    current_number: int
     latest_id: str
```

Comment:
Optional ???

Reply:
I've kept these in in the end, as although they aren't always used in the create method, they are always attributes needed on all the classes, and get populated during the process.